File: cpp-answers/191.位-1-的个数.cpp
/*
 * @Author: your name
 * @Date: 2020-06-06 22:42:23
 * @LastEditTime: 2020-06-06 22:48:51
 * @LastEditors: your name
 * @Description: In User Settings Edit
 * @FilePath: \leetcode刷题\191.位-1-的个数.cpp
 */
/*
 * @lc app=leetcode.cn id=191 lang=cpp
 *
 * [191] 位1的个数
 */

// @lc code=start
class Solution
{
public:
    int hammingWeight(uint32_t n)
    {
        int count = 0;
        while (n)
        {
            n &= (n - 1);
            count++;
        }
        return count;
    }
};
// @lc code=end

File: rust-answers/191.位-1-的个数.rs
/*
 * @lc app=leetcode.cn id=191 lang=rust
 *
 * [191] 位1的个数
 */

// @lc code=start
impl Solution {
    pub fn hammingWeight (mut n: u32) -> i32 {
        // n.count_ones() as i32
        let mut ret = 0;
        while n != 0 {
            n = n & (n - 1);
            ret += 1;
        }
        ret
    }
}
// @lc code=end

